### PR TITLE
Champion Tech Sports E2E Test

### PR DESCRIPTION
**Pull request by Dasariroja**
Description :
Ticket Description

How can this to be tested 
Why : 

Browser: 

Justification :
screenshots 
Ticket Number 

Champion Tech Sports E2E Test

### CODE REVIEW

File: functionaltest/playwright/components/locators/detailslocators.ui.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+readonly skype: Locator;
+readonly clickphone: Locator;
+readonly clickonnumber:Locator;
+readonly methodofcontact: Locator;
+readonly hear: Locator;
+readonly searchenginebtn: Locator;
+readonly productinterest: Locator;
+readonly comments: Locator;
+readonly sent: Locator;
+
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/detailslocators.ui.ts:17]: Remove un wanted spaces

File: functionaltest/playwright/components/locators/cookieslocators.ui.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { Locator, Page } from "@playwright/test"
+
+export class CookiesLocators {
+
+readonly agreebtn : Locator;
+
+constructor(page:Page)
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/cookieslocators.ui.ts]: Use formate document or prettier.

File: functionaltest/playwright/tests/chamtechE2E.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    await detailsmethods.Clickonnumberbtn();
+
+    await detailsmethods.Clickonmethodofcontact();
+    await detailsmethods.Clickonhearabout();
+    await detailsmethods.Clickonasearchengin();
+    await detailsmethods.Clickonbox();
+    await detailsmethods.Clickoncomments();
+    await detailsmethods.ClickonSentBtn();
+
+    await page.waitForTimeout(5000);
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/tests/chamtechE2E.spec.ts:40]: Remove waits and unwanted spaces

File: functionaltest/playwright/components/locators/detailslocators.ui.ts (hunk 1 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/detailslocators.ui.ts:17]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/cookieslocators.ui.ts (hunk 2 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/cookieslocators.ui.ts]: Based on your suggentions comments updated

File: functionaltest/playwright/tests/chamtechE2E.spec.ts (hunk 3 above)
Comment (Dasariroja) [reply functionaltest/playwright/tests/chamtechE2E.spec.ts:40]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/homelocators.ui.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+import { Locator, Page } from "@playwright/test"
+
+export class HomeLocators {
+readonly sportsbookbtn : Locator;
+readonly logo: Locator;
+readonly aboutus: Locator;
+readonly whychamsports: Locator;
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/homelocators.ui.ts:6]: We need space hear

File: functionaltest/playwright/components/locators/homelocators.ui.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+constructor(page:Page){
+    this.logo = page.locator('img').first();
+    this.aboutus = page.getByText('about us').first();
+    this.whychamsports = page.getByText('why champion sports').first();
+    this.sportsbookbtn = page.locator('.btn_list').first();
+}
+}
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/homelocators.ui.ts:14]: Formate document

File: functionaltest/playwright/components/locators/productlocators.ui.ts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import { Locator, Page } from "@playwright/test"
+export class Productlocators {
+readonly demobtn: Locator;
+
+constructor(page:Page){
+    this.demobtn = page.locator('.txt_btn').first();
+}
+}
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/productlocators.ui.ts:8]: Formate documente

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+   {
+    await this.detailslocators.comments.fill('Right Comments here ');
+
+   }
+   async ClickonSentBtn()
+   {
+      await this.detailslocators.sent.click();
+   }
+   }
+
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/methods/detailsmethods.ui.ts]: This file need formate document

File: functionaltest/playwright/components/methods/productmethods.ui.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import { Page } from "@playwright/test"
+import { Productlocators } from "../locators/productlocators.ui";
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/methods/productmethods.ui.ts:2]: One line space needed

File: functionaltest/playwright/tests/chamtechE2E.spec.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
+import { test } from "@playwright/test";
+import { CookiesMethods } from "../components/methods/cookiesmethods.ui";
+import { DetailsMethods } from "../components/methods/detailsmethods.ui";
+import { HomeMethods } from "../components/methods/homemethods.ui";
+import { ProductMethods } from "../components/methods/productmethods.ui";
+
+test.only('Navigate to champion sports', async ({ page }) => {
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/tests/chamtechE2E.spec.ts]: .only needs to remove

File: functionaltest/playwright/components/locators/cookieslocators.ui.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { Locator, Page } from "@playwright/test"
+export class CookiesLocators {
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/cookieslocators.ui.ts:2]: One line space needed hear

File: functionaltest/playwright/components/locators/homelocators.ui.ts [hunk 11]
```diff
@@ -0,0 +1,14 @@
+import { Locator, Page } from "@playwright/test"
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/homelocators.ui.ts:1]: One line space needed hear

File: functionaltest/playwright/components/locators/detailslocators.ui.ts [hunk 12]
```diff
@@ -0,0 +1,33 @@
+import { Locator, Page } from "@playwright/test"
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/detailslocators.ui.ts:1]: One line space needed hear

File: functionaltest/playwright/components/locators/productlocators.ui.ts [hunk 13]
```diff
@@ -0,0 +1,8 @@
+import { Locator, Page } from "@playwright/test"
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/locators/productlocators.ui.ts:1]: One line space

File: functionaltest/playwright/components/methods/cookiesmethods.ui.ts [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import { Page } from "@playwright/test"
+import { CookiesLocators } from "../locators/cookieslocators.ui";
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/methods/cookiesmethods.ui.ts:2]: One line space needed hear

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+import { DetailsLocators } from "../locators/detailslocators.ui";
+
+export class DetailsMethods {
+    readonly detailslocators: DetailsLocators;
+
+    constructor(page: Page) {
+        this.detailslocators = new DetailsLocators(page);
+    }
+
+    async ClickOnFstName() {
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/methods/detailsmethods.ui.ts]: ClickOnFirstName has typo issue `clickOnFirstName`

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts [hunk 16]
```diff
@@ -0,0 +1,66 @@
...
+        await this.detailslocators.firstname.fill('Roja');
+    }
+
+    async clickonlastname() {
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/methods/detailsmethods.ui.ts]: clickonlastname has typo issue `clickOnLastName`

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts [hunk 17]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    async Clickoncomments() {
+        await this.detailslocators.comments.fill('Right Comments here ');
+    }
+
+    async ClickonSentBtn() {
+        await this.detailslocators.sent.click();
+    }
+}
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/methods/detailsmethods.ui.ts:66]: All method name has to change

File: functionaltest/playwright/components/methods/homemethods.ui.ts [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    async ClickonWhyChamSports() {
+        await this.homelocators.whychamsports.click();
+    }
+
+    async ClickonSportsBook() {
+        await this.homelocators.sportsbookbtn.click();
+    }
+}
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/components/methods/homemethods.ui.ts:26]: Same here as all method names should change

File: functionaltest/playwright/tests/chamtechE2E.spec.ts [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+    await homemethods.ClickonLogo();
+    await homemethods.ClickonAboutUsBtn();
+    await homemethods.ClickonWhyChamSports();
+    await homemethods.ClickonLogo();
+    await homemethods.ClickonSportsBook();
+
+    const productmethods = new ProductMethods(page);
+    await productmethods.ClickOnDemoBtn();
+    await page.waitForLoadState();
+    await page.waitForTimeout(30000);
```
Comment (Prasanthgurasala) [inline functionaltest/playwright/tests/chamtechE2E.spec.ts:23]: Why we need to wait here ?

File: functionaltest/playwright/components/locators/homelocators.ui.ts (hunk 4 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/homelocators.ui.ts:6]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/homelocators.ui.ts (hunk 5 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/homelocators.ui.ts:14]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/productlocators.ui.ts (hunk 6 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/productlocators.ui.ts:8]: Based on your suggentions comments updated

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts (hunk 7 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/methods/detailsmethods.ui.ts]: Based on your suggentions comments updated

File: functionaltest/playwright/components/methods/productmethods.ui.ts (hunk 8 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/methods/productmethods.ui.ts:2]: Based on your suggentions comments updated

File: functionaltest/playwright/tests/chamtechE2E.spec.ts (hunk 9 above)
Comment (Dasariroja) [reply functionaltest/playwright/tests/chamtechE2E.spec.ts]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/cookieslocators.ui.ts (hunk 10 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/cookieslocators.ui.ts:2]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/homelocators.ui.ts (hunk 11 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/homelocators.ui.ts:1]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/detailslocators.ui.ts (hunk 12 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/detailslocators.ui.ts:1]: Based on your suggentions comments updated

File: functionaltest/playwright/components/locators/productlocators.ui.ts (hunk 13 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/locators/productlocators.ui.ts:1]: Based on your suggentions comments updated

File: functionaltest/playwright/components/methods/cookiesmethods.ui.ts (hunk 14 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/methods/cookiesmethods.ui.ts:2]: Based on your suggentions comments updated

File: functionaltest/playwright/tests/chamtechE2E.spec.ts (hunk 19 above)
Comment (Dasariroja) [reply functionaltest/playwright/tests/chamtechE2E.spec.ts:23]: Based on your suggentions comments updated

File: functionaltest/playwright/components/methods/homemethods.ui.ts (hunk 18 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/methods/homemethods.ui.ts:26]: Based on your suggentions comments updated

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts (hunk 17 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/methods/detailsmethods.ui.ts:66]: Based on your suggentions comments updated

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts (hunk 16 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/methods/detailsmethods.ui.ts]: Based on your suggentions comments updated

File: functionaltest/playwright/components/methods/detailsmethods.ui.ts (hunk 15 above)
Comment (Dasariroja) [reply functionaltest/playwright/components/methods/detailsmethods.ui.ts]: Based on your suggentions comments updated

**Review by Prasanthgurasala — APPROVED** [review]
Looks good to me

**Pull request merged**